libs/angular-accelerator/src/lib/services/breadcrumb.service.ts

```diff
@@ -1,53 +1,79 @@
-import { Injectable } from '@angular/core'
+import { Injectable, OnDestroy } from '@angular/core'
 import { ActivatedRoute, ActivatedRouteSnapshot, Data, NavigationEnd, ParamMap, Router } from '@angular/router'
 import { TranslateService } from '@ngx-translate/core'
 import { BehaviorSubject, filter } from 'rxjs'
 import { MenuItem } from 'primeng/api'
 import { BreadCrumbMenuItem } from '../model/breadcrumb-menu-item.model'
 
 @Injectable({ providedIn: 'any' })
-export class BreadcrumbService {
+export class BreadcrumbService implements OnDestroy {
   private itemsSource = new BehaviorSubject<MenuItem[]>([])
   generatedItemsSource = new BehaviorSubject<MenuItem[]>([])
 
   itemsHandler = this.itemsSource.asObservable()
 
   constructor(private router: Router, private activeRoute: ActivatedRoute, private translateService: TranslateService) {
-    const breadcrumbs: MenuItem[] = []
-    this.addBreadcrumb(this.activeRoute.snapshot, [], breadcrumbs)
-    this.generatedItemsSource.next(breadcrumbs)
+    this.generateBreadcrumbs(this.activeRoute.snapshot)
     this.router.events.pipe(filter((e) => e instanceof NavigationEnd)).subscribe(() => {
       const root = this.router.routerState.snapshot.root
-      const breadcrumbs: MenuItem[] = []
-      this.addBreadcrumb(root, [], breadcrumbs)
-
-      this.generatedItemsSource.next(breadcrumbs)
+      this.generateBreadcrumbs(root)
     })
   }
+  ngOnDestroy(): void {
```

Comment:
No needed to unsubscribe from the BehavoirSubjects. Unsubscribe from router.events